tool/plugin/lib/plugin.dart

```diff
@@ -624,6 +627,48 @@ class GenerateCommand extends ProductCommand {
 
   SyntheticBuildSpec makeSyntheticSpec(List specs) =>
       SyntheticBuildSpec.fromJson(specs[0], release, specs[2]);
+
+  void generateLiveTemplates() {
+    // Find all the live templates.
+    final templateFragments = Directory(p.join('resources', 'liveTemplates'))
+        .listSync()
+        .where((entity) => entity is File)
+        .where((file) => p.extension(file.path) == 'txt')
+        .cast<File>()
+        .toList();
+    final templateFile =
+        File(p.join('resources', 'liveTemplates', 'flutter_miscellaneous.xml'));
+    var contents = templateFile.readAsStringSync();
+
+    log('writing ${p.relative(templateFile.path)}');
+
+    for (var file in templateFragments) {
+      final name = p.basenameWithoutExtension(file.path);
+
+      var replaceContents = file.readAsStringSync();
+      replaceContents = replaceContents
+          .replaceAll('\n', '&#10;')
+          .replaceAll('<', '&lt;')
+          .replaceAll('>', '&gt;');
+
+      // look for '<template name="$name" value="..."'
+      final regexp = RegExp('<template name="$name" value="([^"]+)"');
+      final match = regexp.firstMatch(contents);
+      if (match == null) {
+        throw 'No entry found for "$name" live template in ${templateFile.path}';
```

Comment:
I don't understand this constraint. It makes it hard to add new live templates. I understand we need to control the size of the pop-up but this means we have to manually add a new one to the xml file.

Reply:
Yeah, this is only here so that we can provide feedback if an entry isn't found during the copy / in-line stage. We don't have enough info in the txt files to create a full xml entry in the live templates file.

When adding a new template, people will need to create a new entry both in the xml file and create a new txt file. I added that into - how to add a new live template - to the readme.